src/main/java/gregtech/common/entities/PowderbarrelEntity.java

```diff
@@ -0,0 +1,102 @@
+package gregtech.common.entities;
+
+import gregtech.api.util.BlockUtility;
+import gregtech.api.util.GregFakePlayer;
+import net.minecraft.block.Block;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.MoverType;
+import net.minecraft.entity.item.EntityItem;
+import net.minecraft.entity.item.EntityTNTPrimed;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+
+import java.util.Collections;
+import java.util.List;
+
+public class PowderbarrelEntity extends EntityTNTPrimed {
+
+    protected float fakeExplosionStrength = 3.5F;
+    protected float range = 2;
+
+    public PowderbarrelEntity(World worldIn, double x, double y, double z, EntityLivingBase igniter) {
+        super(worldIn, x, y, z, igniter);
+    }
+
+    public PowderbarrelEntity(World worldIn) {
+        super(worldIn);
+    }
+
+    @Override
+    public void onUpdate() {
+        this.prevPosX = this.posX;
+        this.prevPosY = this.posY;
+        this.prevPosZ = this.posZ;
+        if (!this.hasNoGravity()) {
+            this.motionY -= 0.03999999910593033D;
+        }
+
+        this.move(MoverType.SELF, this.motionX, this.motionY, this.motionZ);
+        this.motionX *= 0.9800000190734863D;
+        this.motionY *= 0.9800000190734863D;
+        this.motionZ *= 0.9800000190734863D;
+        if (this.onGround) {
+            this.motionX *= 0.699999988079071D;
+            this.motionZ *= 0.699999988079071D;
+            this.motionY *= -0.5D;
+        }
+
+        setFuse(this.getFuse() - 1);
+        if (this.getFuse() <= 0) {
+            this.setDead();
+            if (!this.world.isRemote && !this.inWater) {
+                this.explode();
+            }
+        } else {
+            this.handleWaterMovement();
+            this.world.spawnParticle(EnumParticleTypes.SMOKE_NORMAL, this.posX, this.posY + 0.5D, this.posZ, 0.0D, 0.0D, 0.0D, new int[0]);
+        }
+
+    }
+
+    protected void explode() {
```

Comment:
Explosion removes water blocks even water source block which is kind of strange and maybe not intended. Also similarly it will drop TNT bock placed in range instead of priming it.